matplotlib2tikz/legend.py

```diff
@@ -106,44 +107,74 @@ def draw_legend(data, obj):
 
     if loc == 1:
         # upper right
-        position = None
+        if obj._bbox_to_anchor:
```

Comment:
Another possibility: Override leave the if-else conditional as is, and append conditionally override `position` after that:
```
if obj._bbox_to_anchor:
    position = [bbox_center[0], bbox_center[1]]
```